### DISCUSSION TITLE
npm i @umijs/preset-ui -D  umi创建小米饭 出现镜像源过期

### DISCUSSION THREAD

**Discussion by Abengate** [umijs/umi#12113] [Q&A]
今天突发想用一下umi 小米饭， 但出现了 npm镜像源过期，于是便更换了最新的镜像源

![image](https://github.com/umijs/umi/assets/122261591/d3edec7b-293a-458c-be60-123ca2966b40)

但是当我重新运行的时候，还是报旧镜像源错误。

我删除了node_modules这个模块，在重新安装 还是出现以下问题，yarn.lock中没有旧的镜像源。

这个问题在我创建 ant design pro 项目 使用   yarn add @umijs/preset-ui -D  也会出现同样的问题

![image](https://github.com/umijs/umi/assets/122261591/7811f62e-4fd4-4ee0-a678-1ea309ffbf63)

![image](https://github.com/umijs/umi/assets/122261591/f7d302a3-41c4-47a1-ae7b-d17c4b01141b)

请问各位，这种情况要怎么解决？

**Comment by Abengate** [comment]
[npmmirror-quick](https://www.npmjs.com/package/npmmirror-quick)我尝试了用这个包跑一下更新 lock 文件，但是还是没用

**Reply by Scyng** [reply]
同问，请问最后怎么解决的？

**Comment by fz6m** [comment]
虽然你的 lock 文件里没有旧的镜像源，但是你的全局 npm 配置可能还是旧的，导致你安装新的包还是从旧源下载，所以你得改一下全局 npm 配置：

```bash
  npm config set registry https://registry.npmmirror.com/
  yarn config set registry https://registry.npmmirror.com/
```

另外确保你的 node >= 16 ，npm >= 8 。

现在 umi 和前端社区内更推荐使用 pnpm 管理项目的依赖，使用方式和 npm 一致，通过如下方式安装：

```bash
  npm i -g pnpm
  pnpm -v
  pnpm setup
```

另外你初始化这个项目是 umi 3 的，umi 3 已经不维护了，建议选择创建 umi 4 的模板项目，创建方式：

```bash
  pnpm create umi my-app
```

如果你要用 ant pro 的模板项目，先确保你的 ant pro cli 是最新版本的：

```bash
  npm uninstall -g @ant-design/pro-cli
  yarn global remove @ant-design/pro-cli
  pnpm i -g @ant-design/pro-cli
  pro create myapp
```

**Comment by Abengate** [comment]
找不到解决方法，放弃了




陈鹏
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "umijs/umi"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年5月7日(星期二) 下午3:58
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [umijs/umi] npm i @umijs/preset-ui -D  umi创建小米饭 出现镜像源过期 (Discussion #12113)





 
同问，请问最后怎么解决的？

**Reply by jiajiaxing1** [reply]
看图中你用的get..更换镜像源  有更换成功吗
<img width="942" alt="image" src="https://github.com/umijs/umi/assets/130677115/502a125c-f47d-4e69-bce8-c080a80bec19">

**Comment by fz6m** [comment]
npmmirror-quick 这个工具可以帮你把 lock 文件里的旧 url 替换成 npmmirror ，同时你也要全局设定成 npmmirror ，之后你搜一下 lock 文件里没有旧源的关键词（比如 `taobao`），应该就更新成功了，之后删掉 node_modules 重新安装依赖。

**Comment by mx52jing** [comment]
试一下这个方案呢[11193](https://github.com/ant-design/ant-design-pro/issues/11193#issuecomment-2242299328)
